### ISSUE TITLE
Transaction support broken as of MongoDB.Driver version 2.8.1

### ISSUE THREAD

**Issue by jbogard** [Mongo2Go/Mongo2Go#79]
I believe it's because this check was added:

https://jira.mongodb.org/browse/CSHARP-2561

Just change the test suite from 2.7.0 to 2.8.0, still passes. Change to 2.8.1 - failure!

**Comment by JohannesHoppe** [comment]
I have no account for this Jira. Could you summarize the issue?

Also how does this affect Mongo2Go? It has no external dependencies...

Or did you just referred to the internal test suite of Mongo2Go?!

**Comment by jbogard** [comment]
I started with a minimal repro...then checked the Mongo2Go suite, if you upgrade the MongoDB.Driver dependency to 2.8.1, it starts to break:

```
  Error Message:
   System.NotSupportedException : StartTransaction cannot determine if transactions are supported because there are no connected servers.
  Stack Trace:
     at MongoDB.Driver.Core.Bindings.CoreSession.EnsureTransactionsAreSupported()
   at MongoDB.Driver.Core.Bindings.CoreSession.StartTransaction(TransactionOptions transactionOptions)
   at MongoDbTests.TransactionTests.Test() in C:\users\jbogard\source\repos\MongoDbTests\MongoDbTests\UnitTest1.cs:line 100

Test Run Failed.
Total tests: 1
     Failed: 1
 Total time: 12.2065 Seconds
```

That exception was added with the JIRA issue I linked starting in MongoDB.Driver version 2.8.1. I'll attach a minimal repro, you can make the test pass by downgrading to 2.8.0, then fail by changing to 2.8.1.

See attached repro.

[MongoDbTests.zip](https://github.com/Mongo2Go/Mongo2Go/files/3521353/MongoDbTests.zip)

**Comment by jbogard** [comment]
Found the issue - comes down to Mongo not being able to create collections in a transaction, so you have to do that *outside* of a transaction to get things working properly. False alarm!

**Comment by JohannesHoppe** [comment]
😀👍

**Comment by Manny651** [comment]
I just encountered the same issue. @jbogard I do not see any collection creation in your example. Why do you consider this as a false alarm?

**Comment by jbogard** [comment]
@Manny651 You need to create the collections first, but you can't create collections in a transaction.

**Comment by Manny651** [comment]
@jbogard I don't get it. The exception raises on StartTransaction. How is this related to the collection creation?

**Comment by jbogard** [comment]
Because once I created the collections outside of a transaction then everything works.
